index2.js

```diff
@@ -742,4 +742,41 @@ client.on("messageReactionAdd", (reaction, user) => {
   }
 });
 
+client.on("voiceStateUpdate", (oldUser, newUser) => {
+  const oldChannel = oldUser.channel;
+  const newChannel = newUser.channel;
+
+  if (
+    oldChannel &&
+    (oldChannel.name.startsWith(config.privateChannelPrefix) ||
+      oldChannel.name.endsWith(config.privateChannelSuffix)) &&
+    oldChannel.members.size === 0
+  ) {
+    // User left a channel, and the channel is now empty
+    oldChannel.updateOverwrite(
+      oldChannel.guild.roles.everyone,
+      {
+        VIEW_CHANNEL: false,
+      },
+      "Voice channel made invisible after last user left"
+    );
+  }
+
+  if (
+    newChannel &&
+    (newChannel.name.startsWith(config.privateChannelPrefix) ||
+      newChannel.name.endsWith(config.privateChannelSuffix)) &&
+    newChannel.members.size === 1
+  ) {
+    // User joined a channel, and is the first to join
+    newChannel.updateOverwrite(
+      newChannel.guild.roles.everyone,
+      {
+        VIEW_CHANNEL: true,
```

Comment:
View channel should not be true for everyone - it only should be set to the neutral position.

Reply:
Resolved in e9e53558db772195d46dffa3e5d012fba7d6c847